ksqldb-common/src/main/java/io/confluent/ksql/metrics/MetricAppender.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.metrics;
+
+import com.google.common.collect.ImmutableList;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.SystemTime;
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * {@code MetricAppender} publishes JMX metrics around the number of messages that
+ * are sent to any logger configured to use this appender.
+ */
+public class MetricAppender extends AppenderSkeleton {
+
+  private static final String KSQL_LOGGING_JMX_PREFIX = "io.confluent.ksql.metrics.logging";
+  private static final String KSQL_LOGGING_METRIC_GROUP = "ksql-logging";
+
+  private final Metrics metrics;
+  private final Sensor errors;
+  private final Sensor warns;
+  private final Sensor infos;
+
+  public MetricAppender() {
+    metrics = new Metrics(
+        new MetricConfig().samples(100).timeWindow(1, TimeUnit.SECONDS),
+        ImmutableList.of(new JmxReporter()),
+        new SystemTime(),
+        new KafkaMetricsContext(KSQL_LOGGING_JMX_PREFIX)
+    );
+
+    errors = metrics.sensor(KSQL_LOGGING_METRIC_GROUP + "-error-rate");
+    errors.add(
+        metrics.metricName("errors", KSQL_LOGGING_METRIC_GROUP, "number of error logs per second"),
+        new Rate()
+    );
+
+    warns = metrics.sensor(KSQL_LOGGING_METRIC_GROUP + "-warn-rate");
+    warns.add(
+        metrics.metricName("warns", KSQL_LOGGING_METRIC_GROUP, "number of warn logs per second"),
+        new Rate()
+    );
+
+    infos = metrics.sensor(KSQL_LOGGING_METRIC_GROUP + "-info-rate");
```

Comment:
INFO logs may be a lot. Wonder if the `.record` method can handle integer overflow

Reply:
Just checked this. I think we increment a double for value and an event count internally which is a long. 
https://github.com/confluentinc/ce-kafka/blob/master/clients/src/main/java/org/apache/kafka/common/metrics/stats/WindowedSum.java#L37 and 
https://github.com/confluentinc/ce-kafka/blob/master/clients/src/main/java/org/apache/kafka/common/metrics/stats/SampledStat.java#L52

Reply:
I checked that `event_count` gets reset periodically. Its just the double value that is concerning. I sampled the count of all logs for the last 15 mins. And It was close to 10 million. So like a billion info logs in a day. I guess this should be fine for the metric.

Reply:
I had the same thought but I think we should be fine - also we can remove INFO if necessary, I added that because I have seen incidents where a large increase in any kind of logging is concerning, but ERROR/WARN are most important.